src/kbmod/image_collection.py

```diff
@@ -0,0 +1,510 @@
+    Attributes which values may or may not be availible have their own getter
+    methods (f.e. `get_standardizer`, `get_wcs` etc.).
+
+    Some standardizers, such as `MultiExtensionfits`, map a single location and
+    timestamp to potentially many individual extensions each with a slightly
+    different on-sky location. Therefore the list of the standardizers does
+    not neccessarily match the length of the metadata table. Metadata table is
+    the metadata per extension, unravelled into a flat table.
+
+    Parameters
+    ----------
+    metadata : `~astropy.table.Table`
+        A table of exposure metadata properties
+    serializers : `list`
+        A list of valid serializer names, used to extract the metadata in its
+        row.
+
+    Attributes
+    ----------
+    data : `~astropy.table.Table`
+        Table of exposure metadata properties.
+    standardizers : `list`
+        Standardizer names used to standardize the rows of the metadata.
+        The names are guaranteed to exist, but the standardizer object may not
+        be availible.
+    wcs : `list`
+        List of `~astropy.wcs.WCS` objects that correspond to the rows
+        of the metadadata.
+    bbox : `list`
+        List of `dict` objects containing the pixel and on-sky coordinates of
+        the central and corner (0, 0) pixel.
+
+    Raises
+    ------
+    ValueError :
+        when instantiated from a Table which does not have the required
+        columns, or has null-values in the required columns.
+    """
+
+    required_metadata = ["location", "mjd", "ra", "dec"]
+    _supporting_metadata = ["std_name", "std_idx", "ext_idx", "wcs", "bbox", "config"]
+
+    ########################
+    # CONSTRUCTORS
+    ########################
+    def _validate(self, metadata):
+        """Validates the required metadata exist and is not-null.
+
+        Required metadata is the ``location`` of the target, ``mjd``, ``ra``,
+        ``dec`` and the standardizer-row lookup indices ``std_idx`` and
+        ``ext_idx``.
+
+        Parameters
+        ----------
+        metadata : `~astropy.table.Table`
+            Astropy Table containing the required metadata.
+
+        Returns
+        -------
+        valid : `bool`
+            When ``True`` the metadata is valid, when ``False`` it isn't.
+        message: `str`
+            Validation failure message, if any.
+        """
+        if not isinstance(metadata, Table):
+            return False, "not an Astropy Table object."
+
+        # if empty table
+        if not metadata:
+            return False, "an emtpy table."
+
+        cols = metadata.columns
+        missing_keys = [key for key in self.required_metadata if key not in cols]
+        if missing_keys:
+            return False, f"missing required columns: {missing_keys}"
+
+        # check that none of the actual required column entries are empty in
+        # some way. perhaps we should be checking np.nan too?
```

Comment:
We are very inconsistent about checking for NaNs throughout KBMOD, but we should.

Reply:
Should I open an issue?

Reply:
Yes. Let's do that. It's a good first issue.